datalad/cmd.py

```diff
@@ -1323,9 +1324,27 @@ def close(self, return_stderr=False):
                 "Closing stdin of %s and waiting process to finish", process)
             process.stdin.close()
             process.stdout.close()
-            process.wait()
+            # try waiting for the annex process to finish 3 times for 3 sec
+            # with 1s pause in between
+            try:
+                try_multiple(
+                    # ntrials
+                    3,
+                    # exception to catch
+                    subprocess.TimeoutExpired,
+                    # base waiting period
+                    1.0,
+                    # function to run
+                    process.wait,
+                    timeout=3.0,
+                )
+            except subprocess.TimeoutExpired:
+                lgr.warning(
+                    "Batched process %s did not finish, abandoning it without killing it",
+                    process)
+            if process.returncode:
```

Comment:
why it is conditioned on unsuccessful (success is 0 right?) return code now?  or was it intended to be `if not process.returncode`?

Reply:
True, it should be `if ... is None` that signals a running process

Reply:
I amended the respective commit with a fixup -- thx for noticing!